articles/tutorials/04_working_with_textures/index.md

```diff
@@ -0,0 +1,364 @@
+
+```cs
+_spriteBatch.Draw(_logo, 
+  new Vector2(Window.ClientBounds.Width, Window.ClientBounds.Height) * 0.5f,
+  null,
+  Color.White,
+  MathHelper.ToRadians(90),
+  new Vector2(_logo.Width, _logo.Height) * 0.5f,
+  1.0f,
+  SpriteEffects.None,
+  0.0f);
+```
+
+In the code above, we have adjust the *origin* parameter to use the center point of the texture. This will be the origin point at which it will now rotate from. It is also the origin point at which it is rendered from, so we no longer need to subtract half the logo width and height from the *position* parameter. After making this change, if we run the game now, we'll see it drawn correctly rotated 90° and centered on the game window.
+
+<figure><img src="./images/logo-rotated-centered.png" alt="Figure 4-11: The MonoGame logo drawn rotated 90° and centered on the game window."><figcaption><p><strong>Figure 4-11: The MonoGame logo drawn rotated 90° and centered on the game window.</strong></p></figcaption></figure>
+
+## Drawing Texture Regions
+
+Above, we saw the full parameter list for drawing a texture using `SpriteBatch.Draw`. One of those parameters was called *sourceRectangle*. So far, we've just set this parameter to `null`, which specifies that the full texture should be rendered. However, we can make use of the *sourceRectangle* parameter to specify a region within the texture itself to draw instead of drawing the full texture.
+
+For instance, take the logo image we've been using. We can break it down into two distinct regions; the logo and the MonoGame word mark.
+
+<figure><img src="./images/logo-texture-regions.png" alt="Figure 4-12: The MonoGame logo broken down into texture regions."><figcaption><p><strong>Figure 4-12: The MonoGame logo broken down into texture regions.</strong></p></figcaption></figure>
+
+We can see from this image that the actual logo starts at position (0, 0) and is 128px wide and 128px tall. Likewise, the MonoGame word mark starts at position (150, 34) and is 458px wide and 58px tall. Knowing the starting position and the width and height of the region gives us a defined rectangle that we can use as the *sourceRectangle*.
+
+Let's see this in action by only drawing the MonoGame logo icon part of the texture. First, after the call to the `Clear` method, add the following variable:
+
+```cs
+Rectangle iconSourceRect = new Rectangle(0, 0, 128, 128);
+```
+
+Next, update the `_spriteBatch.Draw` method call to the following
+
+```cs
+_spriteBatch.Draw(_logo,

```

Comment:
Add comments to each parameter

Reply:
Trying to find a way to add comments to the parameters without it looking messy.  Which option would you suggest (or a different approach if you have one)

**over each parameter**
```cs
_spriteBatch.Draw(
  // texture
  _logo, 
  // position
  new Vector2(
    (Window.ClientBounds.Width * 0.5f) - (_logo.Width * 0.5f),
    (Window.ClientBounds.Height * 0.5f) - (_logo.Height * 0.5f)),
  // sourceRectangle
  null,
  // color
  Color.White,
  // rotation
  0.0f,
  // origin
  Vector2.Zero,
  // scale
  1.0f,
  // effects
  SpriteEffects.None,
  // layerDepth
  0.0f);
```

**Right-aligned after parameter**
```cs
_spriteBatch.Draw(
  _logo,              // texture
  new Vector2(        // position
    (Window.ClientBounds.Width * 0.5f) - (_logo.Width * 0.5f),
    (Window.ClientBounds.Height * 0.5f) - (_logo.Height * 0.5f)),
  null,               // sourceRectangle
  Color.White,        // color
  0.0f,               // rotation
  Vector2.Zero,       // origin
  1.0f,               // scale
  SpriteEffects.None, // effects
  0.0f);              // layerDepth
```

**Left-aligned before parameter**
```cs
_spriteBatch.Draw(
  /* texture */         _logo, 
  /* position */        new Vector2(
                          (Window.ClientBounds.Width * 0.5f) - (_logo.Width * 0.5f),
                          (Window.ClientBounds.Height * 0.5f) - (_logo.Height * 0.5f)),
  /* sourceRectangle */ null,
  /* color */           Color.White,
  /* rotation */        0.0f,
  /* origin */          Vector2.Zero,
  /* scale */           1.0f,
  /* effects */         SpriteEffects.None, 
  /* layerDepth */      0.0f);
```

**Using named parameters instead of comments**
```cs
_spriteBatch.Draw(
  texture: _logo, 
  position: new Vector2(
    (Window.ClientBounds.Width * 0.5f) - (_logo.Width * 0.5f),
    (Window.ClientBounds.Height * 0.5f) - (_logo.Height * 0.5f)),
  sourceRectangle: null,
  color: Color.White,
  rotation: 0.0f,
  origin: Vector2.Zero,
  scale: 1.0f,
  effects: SpriteEffects.None, 
  layerDepth: 0.0f);
```